dvc/remote/s3.py

```diff
@@ -207,10 +208,24 @@ def _list_paths(self, path_info, max_items=None):
     def list_cache_paths(self):
         return self._list_paths(self.path_info)
 
+    def isfile(self, path_info):
+        if path_info.path.endswith("/"):
+            return False
+
+        try:
+            self.s3.head_object(Bucket=path_info.bucket, Key=path_info.path)
+        except ClientError:
```

Comment:
You still need to check an error code, see https://www.peterbe.com/plog/fastest-way-to-find-out-if-a-file-exists-in-s3

Reply:
@Suor, that one is outdated, I ran the benchmarks last month and it wasn't the case